examples/end-to-end-session-based/03-Session-based-Yoochoose-multigpu-training-PyT.ipynb

```diff
@@ -0,0 +1,318 @@
+{
```

Comment:
<ul><li><strong>specifying multiple GPUs</strong></li></ul><p>I believe that when working with <code>DistributedDataParallel</code> the <code>CUDA_VISIBLE_DEVICES</code> env variable is not really considered by <code>torch.distributed.launch</code>, as the launcher spawns one process per GPU (which is provided via the <code>--local_rank</code> arg) . The number of GPUs is defined by the <code>--nproc_per_node</code> argument.</p><p> </p><p>You say <code>using different batches of the data in a model-parallel fashion</code>,but that is in fact <code>data -parallel fashion</code>.</p><p> </p><ul><li><strong>data repartitioning</strong>:</li></ul><p>You could link here to this <a href="https://nvidia-merlin.github.io/Transformers4Rec/main/multi_gpu_train.html#distributeddataparallel" target="_blank">doc</a> that explains how parquet data can be partitioned when saving</p><p> </p><ul><li>I think we need to provide instructions on how to download the data required to run this example (maybe pointing to the example notebook that explains that)</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/Transformers4Rec/pull/521/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>